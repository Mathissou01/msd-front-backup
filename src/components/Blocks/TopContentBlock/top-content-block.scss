@use "src/styles/01-settings/settings.colors" as colors;
@use "src/styles/02-tools/tools.typography" as typo;
@use "src/styles/02-tools/tools.inputs" as input;
@use "src/styles/02-tools/tools.media-queries" as media;

.c-TopContentBlock {
  display: flex;
  flex-direction: column;
  gap: 24px;
  align-items: center;
  padding: 40px 16px;

  @include media.from-tablet {
    padding: 80px 124px;
  }

  &__RowCardsBlock {
    display: flex;
    flex-direction: column;
    gap: 24px;
    padding: 0 16px;

    @include media.from-tablet {
      flex-direction: row;
      width: 903px;
      padding: 0;
    }
  }
}
