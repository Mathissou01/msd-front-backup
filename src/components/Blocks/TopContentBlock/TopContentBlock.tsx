import React from "react";
import CommonBlockHeading from "../../Common/CommonBlockHeading/CommonBlockHeading";
import CommonButton from "../../Common/CommonButton/CommonButton";
import {
  GetNewestTopContentsQuery,
  TopContentBlockEntity,
} from "../../../graphql/codegen/generated-types";
import CommonTopContentCard from "../../Common/CommonTopContentCard/CommonTopContentCard";
import "./top-content-block.scss";
import CommonCardBlock from "../../Common/CommonCardBlock/CommonCardBlock";

interface ITopContentBlock {
  data: TopContentBlockEntity;
  newestTopContents: GetNewestTopContentsQuery;
}
export default function TopContentBlock({
  data,
  newestTopContents,
}: ITopContentBlock) {
  /* Static Data */
  const defaultImage = "/images/images-temp/newsDesktop.jpg";
  const defaultImageMobile = "/images/images-temp/newsMobile.png";

  // TODO: temporarily static data, replace with real tags later
  const labelButton = "Voir plus d’actualités et d’événements";
  const defaultHref = "/";

  /* Local Data */
  const titleContent = data?.attributes?.titleContent ?? "";
  const hasTopContent = data.attributes?.hasTopContent;
  const displayLastThreeContents = data.attributes?.displayLastThreeContents;
  const contentTopNewsOrEvent =
    data?.attributes?.topContent?.data?.attributes?.news?.data?.attributes;
  const titleNews = contentTopNewsOrEvent?.title ?? "";
  const descriptionNews = contentTopNewsOrEvent?.shortDescription ?? "";
  const date = contentTopNewsOrEvent?.publishedAt ?? "";
  const threeTopContents = newestTopContents?.getNewestTopContents;

  return (
    <section className="c-TopContentBlock">
      <CommonBlockHeading titleContent={titleContent} isAlignLeft={true} />
      {hasTopContent && (
        <CommonTopContentCard
          title={titleNews}
          redirectUrl={defaultHref}
          description={descriptionNews}
          imageUrlDesktop={
            contentTopNewsOrEvent?.image?.data?.attributes?.url ?? defaultImage
          }
          imageUrlMobile={
            contentTopNewsOrEvent?.image?.data?.attributes?.url ??
            defaultImageMobile
          }
          date={date}
        />
      )}
      {displayLastThreeContents && (
        <div className="c-TopContentBlock__RowCardsBlock">
          {threeTopContents?.map((topContent, index) => (
            <CommonCardBlock
              key={index}
              title={topContent?.title ?? ""}
              description={topContent?.shortDescription}
              date={topContent?.publishedAt}
              imageUrl={defaultImageMobile}
              href={defaultHref}
            />
          ))}
        </div>
      )}
      <CommonButton label={labelButton} style="primary" />
    </section>
  );
}
