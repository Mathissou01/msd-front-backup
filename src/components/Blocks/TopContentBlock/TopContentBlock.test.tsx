import { render, screen } from "@testing-library/react";
import {
  defaultMockMinData,
  defaultMockFullData,
  newestTopContentMockData,
} from "../../../../__mocks__/TopContentBlockMockData";
import TopContentBlock from "./TopContentBlock";

describe("TopContentBlock", () => {
  it("renders titleContent", () => {
    const { container } = render(
      <TopContentBlock
        data={defaultMockMinData}
        newestTopContents={newestTopContentMockData}
      />,
    );
    const titleContent = screen.getByText("Actualités & Evénements");
    expect(titleContent).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("renders top Content and three topContents", () => {
    const { container } = render(
      <TopContentBlock
        data={defaultMockFullData}
        newestTopContents={newestTopContentMockData}
      />,
    );

    const title = screen.getByText("Lorem ipsum dolor sit amet");
    const shortDescription = screen.getByText(
      "consectetur adipiscing elit. Aliquam facilisis tincidunt est",
    );
    const titleEventOrNews = screen.getByText("Suspendisse et est sem");
    const shortDescriptionEventOrNews = screen.getByText(
      "Vestibulum cursus eros ut ligula lobortis auctor. Quisque luctus sagittis tellus",
    );

    expect(title).toBeInTheDocument();
    expect(shortDescription).toBeInTheDocument();
    expect(titleEventOrNews).toBeInTheDocument();
    expect(shortDescriptionEventOrNews).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
