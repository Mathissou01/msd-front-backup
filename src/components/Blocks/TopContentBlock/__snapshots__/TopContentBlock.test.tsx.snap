// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopContentBlock renders titleContent 1`] = `
<div>
  <section
    class="c-TopContentBlock"
  >
    <div
      class="c-CommonBlockHeading c-CommonBlockHeading_alignLeft"
    >
      <h2>
        Actualités & Evénements
      </h2>
      <div
        class="c-CommonBlockHeading__Line"
      />
    </div>
    <button
      class="c-CommonButton c-CommonButton_primary c-CommonButton_fontSmall"
      type="button"
    >
      Voir plus d’actualités et d’événements
    </button>
  </section>
</div>
`;

exports[`TopContentBlock renders top Content and three topContents 1`] = `
<div>
  <section
    class="c-TopContentBlock"
  >
    <div
      class="c-CommonBlockHeading c-CommonBlockHeading_alignLeft"
    >
      <h2>
        Actualités & Evénements
      </h2>
      <div
        class="c-CommonBlockHeading__Line"
      />
    </div>
    <div
      class="c-CommonTopContentCard"
    >
      <div
        class="c-CommonTopContentCard__Image"
      >
        <img
          alt=""
          class="c-CommonTopContentCard__Img_desktop"
          data-nimg="1"
          decoding="async"
          height="308"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Fimages-temp%2FnewsDesktop.jpg&w=1080&q=75"
          srcset="/_next/image?url=%2Fimages%2Fimages-temp%2FnewsDesktop.jpg&w=640&q=75 1x, /_next/image?url=%2Fimages%2Fimages-temp%2FnewsDesktop.jpg&w=1080&q=75 2x"
          style="color: transparent;"
          width="480"
        />
        <img
          alt=""
          class="c-CommonTopContentCard__Img_mobile"
          data-nimg="1"
          decoding="async"
          height="200"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Fimages-temp%2FnewsMobile.png&w=640&q=75"
          srcset="/_next/image?url=%2Fimages%2Fimages-temp%2FnewsMobile.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2Fimages-temp%2FnewsMobile.png&w=640&q=75 2x"
          style="color: transparent;"
          width="312"
        />
      </div>
      <div
        class="c-CommonTopContentCard__ContentContainer"
      >
        <div
          class="c-CommonTopContentCard__Content"
        >
          <div
            class="c-CommonTopContentCard__ContentTags"
          >
            <div
              class="c-CommonTopContentCard__Tags"
            />
            <div
              class="c-CommonTopContentCard__ContentCardDate"
            >
              5 décembre 2022
            </div>
          </div>
          <div
            class="c-CommonTopContentCard__Description"
          >
            <h2
              class="c-CommonTopContentCard__TitleContent"
            >
              Lorem ipsum dolor sit amet
            </h2>
            <p
              class="c-CommonTopContentCard__DescriptionContent"
            >
              consectetur adipiscing elit. Aliquam facilisis tincidunt est
            </p>
          </div>
          <a
            class="c-CommonTopContentCard__Link"
            href="/"
          >
            <span
              class="c-CommonTopContentCard__Label"
            >
              En savoir plus
            </span>
            <div />
          </a>
        </div>
      </div>
    </div>
    <div
      class="c-TopContentBlock__RowCardsBlock"
    >
      <div
        class="c-CommonCardBlock"
      >
        <div
          class="c-CommonCardBlock__Image"
        >
          <img
            alt=""
            data-nimg="1"
            decoding="async"
            height="220"
            loading="lazy"
            src="/_next/image?url=%2Fimages%2Fimages-temp%2FnewsMobile.png&w=750&q=75"
            srcset="/_next/image?url=%2Fimages%2Fimages-temp%2FnewsMobile.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2Fimages-temp%2FnewsMobile.png&w=750&q=75 2x"
            style="color: transparent;"
            width="343"
          />
        </div>
        <div
          class="c-CommonCardBlock__Content"
        >
          <div
            class="c-CommonCardBlock__ContentHeader"
          >
            <div
              class="c-CommonCardBlock__Tags"
            />
            <div
              class="c-CommonCardBlock__Date"
            >
              29 décembre 2022
            </div>
          </div>
          <div
            class="c-CommonCardBlock__ContentBody"
          >
            <h3
              class="c-CommonCardBlock__Title"
            >
              Suspendisse et est sem
            </h3>
            <p
              class="c-CommonCardBlock__Description"
            >
              Vestibulum cursus eros ut ligula lobortis auctor. Quisque luctus sagittis tellus
            </p>
          </div>
          <a
            class="c-CommonCardBlock__Link"
            href="/"
          >
            <span
              class="c-CommonCardBlock__Label"
            >
              En savoir plus
            </span>
            <div />
          </a>
        </div>
      </div>
      <div
        class="c-CommonCardBlock"
      >
        <div
          class="c-CommonCardBlock__Image"
        >
          <img
            alt=""
            data-nimg="1"
            decoding="async"
            height="220"
            loading="lazy"
            src="/_next/image?url=%2Fimages%2Fimages-temp%2FnewsMobile.png&w=750&q=75"
            srcset="/_next/image?url=%2Fimages%2Fimages-temp%2FnewsMobile.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2Fimages-temp%2FnewsMobile.png&w=750&q=75 2x"
            style="color: transparent;"
            width="343"
          />
        </div>
        <div
          class="c-CommonCardBlock__Content"
        >
          <div
            class="c-CommonCardBlock__ContentHeader"
          >
            <div
              class="c-CommonCardBlock__Tags"
            />
            <div
              class="c-CommonCardBlock__Date"
            >
              5 décembre 2022
            </div>
          </div>
          <div
            class="c-CommonCardBlock__ContentBody"
          >
            <h3
              class="c-CommonCardBlock__Title"
            >
              Café préparation
            </h3>
            <p
              class="c-CommonCardBlock__Description"
            >
              Participer à un café réparation avec La Recyclade au café restaurant de la Maison Phare. Vous apprendrez à faire un diagnostic et à réparer votre petit appareil électrique ou électronique de la vie quotidienne
            </p>
          </div>
          <a
            class="c-CommonCardBlock__Link"
            href="/"
          >
            <span
              class="c-CommonCardBlock__Label"
            >
              En savoir plus
            </span>
            <div />
          </a>
        </div>
      </div>
      <div
        class="c-CommonCardBlock"
      >
        <div
          class="c-CommonCardBlock__Image"
        >
          <img
            alt=""
            data-nimg="1"
            decoding="async"
            height="220"
            loading="lazy"
            src="/_next/image?url=%2Fimages%2Fimages-temp%2FnewsMobile.png&w=750&q=75"
            srcset="/_next/image?url=%2Fimages%2Fimages-temp%2FnewsMobile.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2Fimages-temp%2FnewsMobile.png&w=750&q=75 2x"
            style="color: transparent;"
            width="343"
          />
        </div>
        <div
          class="c-CommonCardBlock__Content"
        >
          <div
            class="c-CommonCardBlock__ContentHeader"
          >
            <div
              class="c-CommonCardBlock__Tags"
            />
            <div
              class="c-CommonCardBlock__Date"
            >
              5 décembre 2022
            </div>
          </div>
          <div
            class="c-CommonCardBlock__ContentBody"
          >
            <h3
              class="c-CommonCardBlock__Title"
            >
              Café préparation
            </h3>
            <p
              class="c-CommonCardBlock__Description"
            >
              Participer à un café réparation avec La Recyclade au café restaurant de la Maison Phare. Vous apprendrez à faire un diagnostic et à réparer votre petit appareil électrique ou électronique de la vie quotidienne
            </p>
          </div>
          <a
            class="c-CommonCardBlock__Link"
            href="/"
          >
            <span
              class="c-CommonCardBlock__Label"
            >
              En savoir plus
            </span>
            <div />
          </a>
        </div>
      </div>
    </div>
    <button
      class="c-CommonButton c-CommonButton_primary c-CommonButton_fontSmall"
      type="button"
    >
      Voir plus d’actualités et d’événements
    </button>
  </section>
</div>
`;
